client/app/telecom/pack/xdsl/modem/templates/config/pack-xdsl-modem-templates-config.controller.js

```diff
@@ -0,0 +1,261 @@
+    }
+
+    if (this.template.detail.WLAN) {
+      // Add ignore column to set parameters to ignore
+      this.template.detail.WLAN.forEach((row) => {
+        const line = row;
+        line.ignore = false;
+        if (this.parametersToIgnore.WLANList && this.parametersToIgnore.WLANList.length > 0) {
+          if (this.parametersToIgnore.WLANList.includes(line.wifiName)) {
+            line.ignore = true;
+          }
+        }
+      });
+    }
+
+    if (this.template.detail.portMapping) {
+      this.template.detail.portMapping.forEach((row) => {
+        const line = row;
+        line.ignore = false;
+        if (this.parametersToIgnore.portMappingList
+            && this.parametersToIgnore.portMappingList.length > 0) {
+          if (this.parametersToIgnore.portMappingList.includes(line.name)) {
+            line.ignore = true;
+          }
+        }
+      });
+    }
+  }
+
+  convertCapabilities(capabilities, paramsToIgnore, isParametersToIgnore) {
+    const parametersToIgnore = {
+      mtuSize: capabilities.mtuSize ? !(capabilities.mtuSize) : '',
+      dmzIP: capabilities.dmzIP ? !(capabilities.dmzIP) : '',
+      LANandDHCP: capabilities.LANandDHCP ? !(capabilities.LANandDHCP) : '',
+    };
+    this.isDisabled = {
+      mtuSize: !capabilities.mtuSize,
+      dmzIP: !capabilities.dmzIP,
+      LANandDHCP: !capabilities.LANandDHCP,
+    };
+    if (isParametersToIgnore) {
+      if (paramsToIgnore) {
+        this.parametersToIgnore = paramsToIgnore;
+      } else {
+        this.parametersToIgnore = parametersToIgnore;
+      }
+    }
+    return parametersToIgnore;
+  }
+
+  displayModal(title, question) {
+    const modal = this.$uibModal.open({
+      animation: true,
+      templateUrl: 'app/telecom/pack/xdsl/modem/templates/config/modal/pack-xdsl-modem-template-config-modal.html',
+      controller: 'XdslModemTemplateConfigModalCtrl',
+      controllerAs: '$ctrl',
+      resolve: {
+        data: () => ({
+          title,
+          question,
+        }),
+      },
+    });
+
+    return modal;
+  }
+
+  deleteSelectedTemplate(line) {
+    const modal = this.displayModal(this.$translate.instant('xdsl_modem_template_config_modal_title_delete'),
+      this.$translate.instant('xdsl_modem_template_config_modal_question_delete', { templateName: line.name }));
+
+    modal.result.then((res) => {
+      if (res === 'confirm') {
+        return this.OvhApiXdsl.TemplateModem().v6().deleteTemplate({
+          name: line.name,
+        })
+          .$promise
+          .then(() => this.fetchTemplates().then((data) => { this.templates = data; }))
+          .then(() => this.TucToast.success(this.$translate.instant('xdsl_modem_template_config_delete_success')))
+          .catch((error) => {
+            this.TucToast.error(this.$translate.instant('xdsl_modem_template_config_delete_error'));
+            return this.$q.reject(error);
```

Comment:
If it's not catched afterward on the promise chain then rejecting the error from here do not have any benefit.